### PR TITLE
Refactor course

### LINKED ISSUE #182: Refactor course module
Create a module for courses and place all the required files into the module

### PR DESCRIPTION
**Pull request by aadiraju**
## Description
Made course module and moved all components into the module

## Types of changes
What types of changes does your code introduce?
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Refactor (only improve the code quality no change in functionality)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
- [x] Issue has been created
- [x] Every file touched is linted
- [ ] Every file touched has tests that cover all the funcitonality with green coverage
- [ ] Documentation is updated (if applicable).

## Issue
closes #182 

## Tests
Please describe in detail what tests you added or changed.

### CODE REVIEW

File: src/app/course/course-routing.module.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        canActivate: [AuthGuard]
+    },
+    {
+        path: 'view/:courseId',
+        component: CourseComponent,
+        canActivate: [AuthGuard]
+    },
+    {
+        path: ':courseId/new-event',
+        component: CourseEventCreateEditComponent,
```
Comment (keyvankhademi) [inline src/app/course/course-routing.module.ts:29]: Please organize these paths. All the ones start with `:courseId` should be nested.

File: src/app/course/course-routing.module.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        component: CourseListComponent,
+        canActivate: [AuthGuard]
+    },
+    {
+        path: 'register/:courseId',
+        component: CourseRegisterComponent,
+        canActivate: [AuthGuard]
+    },
+    {
+        path: 'view/:courseId',
```
Comment (keyvankhademi) [inline src/app/course/course-routing.module.ts:23]: this should be the parent route. We don't need `view` in the url.

File: src/app/course/course-routing.module.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+import {ProblemViewComponent} from "@app/problems/problem-view/problem-view.component";
+
+const routes = [
+    {
+        path: '',
+        component: CourseListComponent,
+        canActivate: [AuthGuard]
+    },
+    {
+        path: 'register/:courseId',
```
Comment (keyvankhademi) [inline src/app/course/course-routing.module.ts:18]: it's better to have courseid/register.

**Review by keyvankhademi — CHANGES_REQUESTED** [review]

**Review by keyvankhademi — APPROVED** [review]

**Comment by keyvankhademi** [comment]
Can you please resolve the conflicts?

**Comment by aadiraju** [comment]
Conflicts resolved

**Review by keyvankhademi — APPROVED** [review]

**Pull request merged**